package-configuration/linux-cli/get_cloudify.py

```diff
@@ -0,0 +1,383 @@
+# tar -xzvf Python-2.7.6.tgz
+# cd Python-2.7.6
+# ./configure --prefix=/usr/local && make && make altinstall
+# curl -L https://www.dropbox.com/s/ibwdmqhwnf4bewc/get-cloudify.py -o get-cloudify.py && python2.7 get-cloudify.py --pythonpath=python2.7 -f # NOQA
+
+# Install Cloudify on Windows (Python 32/64bit)
+# Install Python 2.7.x 32/64bit from https://www.python.org/downloads/release/python-279/  # NOQA
+# Make sure that when you install, you choose to add Python to the system path.
+# Download the script to any directory
+# Run python get-cloudify.py -f
+
+
+import sys
+import subprocess
+import argparse
+import platform
+import os
+import urllib
+import struct
+
+
+DESCRIPTION = '''This script attempts(!) to install Cloudify's CLI on Linux,
+Windows (with Python32 AND 64), and OS X (Darwin).
+On the linux front, it supports Debian/Ubuntu, CentOS/RHEL and Arch.
+
+Installations are supported for both system python and virtualenv
+(using the --virtualenv flag).
+
+If you're already running the script from within a virtualenv and you're not
+providing a --virtualenv path, Cloudify will be installed within the virtualenv
+you're in.
+
+Passing the --wheelspath allows for an offline installation of Cloudify
+from predownloaded Cloudify dependency wheels. Note that if wheels are found
+within the default wheels directory or within --wheelspath, they will (unless
+the --forceonline flag is set) be used instead of performing an online
+installation.
+
+A --nosudo flag can be supplied (If not on Windows) so that prerequisites can
+be installed on machines/containers without the sudo execuable
+(must be run by root user). Sudo for relevant prerequisites is on by default.
+
+By default, the script assumes that the Python executable is in the
+path and is called 'Python' on Linux and 'c:\python27\python.exe on Windows.
+The Python path can be overriden by using the --pythonpath flag.
+
+The script will attempt to install all necessary requirements including
+python-dev and gcc (for Fabric on Linux), pycrypto (for Fabric on Windows),
+pip and virtualenv depending on the OS and Distro you're running on.
+
+Please refer to Cloudify's documentation at http://getcloudify.org for
+additional information.'''
+
+QUIET = False
+VERBOSE = False
+# TODO: put these in a private storage
+PIP_URL = 'https://bootstrap.pypa.io/get-pip.py'
+PYCR64_URL = 'http://www.voidspace.org.uk/downloads/pycrypto26/pycrypto-2.6.win-amd64-py2.7.exe'  # NOQA
+PYCR32_URL = 'http://www.voidspace.org.uk/downloads/pycrypto26/pycrypto-2.6.win32-py2.7.exe'  # NOQA
+
+
+def prn(what):
+    """Print, yo! Oh wait.. unless QUIET.
+    """
+    if QUIET:
+        return
+    print(what)
+
+
+def _print_proc_out(proc):
+    stdout_line = proc.stdout.readline()
+    if len(stdout_line) > 0:
+        prn('STDOUT: {0}'.format(stdout_line))
+
+
+def run(cmd, sudo=False):
+    """This will execute a command either sudo-ically or not.
+    """
+    cmd = 'sudo {0}'.format(cmd) if sudo else cmd
```

Comment:
this sudo logic is redundant.
if user want to run this script with sudo, he will run `sudo get_cloudify.py` just as `sudo get_pip.py`.